lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,194 @@
+# POWERSHELL_COMMON
+
+Set-StrictMode -Version 2
+
+function Compare-XmlDocs($actual, $expected) {
+    if ($actual.Name -ne $expected.Name) {
+        throw "Actual name not same as expected: actual=" + $actual.Name + ", expected=" + $expected.Name
+    }
+    ##attributes...
+
+    if ($actual.Attributes -ne $null -and $expected.Attributes -ne $null) {
+        if ($actual.Attributes.Count -ne $expected.Attributes.Count) {
+           throw "attribute mismatch for actual=" + $actual.Name
+        }
+        for ($i=0;$i -lt $expected.Attributes.Count; $i =$i+1) {
+            if ($expected.Attributes[$i].Name -ne $actual.Attributes[$i].Name) {
+                throw "attribute name mismatch for actual=" + $actual.Name
+            }
+            if ($expected.Attributes[$i].Value -ne $actual.Attributes[$i].Value) {
+                throw "attribute value mismatch for actual=" + $actual.Name
+            }
+        }
+    }
+
+    ##children
+    if ($expected.ChildNodes.Count -ne $actual.ChildNodes.Count)  {
+        throw "child node mismatch. for actual=" + $actual.Name
+    }
+
+    for ($i=0;$i -lt $expected.ChildNodes.Count; $i =$i+1) {
+        if (-not $actual.ChildNodes[$i]) {
+            throw "actual missing child nodes. for actual=" + $actual.Name
+        }
+        Compare-XmlDocs $expected.ChildNodes[$i] $actual.ChildNodes[$i]
+    }
+
+    if ($expected.InnerText) {
+        if ($expected.InnerText -ne $actual.InnerText) {
+           throw "inner text mismatch for actual=" + $actual.Name
+        }
+    }
+    elseif ($actual.InnerText) {
+        throw "actual has inner text but expected does not for actual=" + $actual.Name
+    }
+}
+
+function BackupFile($path) {
+	$backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+	Copy-Item $path $backuppath;
+	return $backuppath;
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$debug_level = Get-AnsibleParam -obj $params -name "_ansible_verbosity" -type "int"
+$debug = $debug_level -gt 2
+
+$dest = Get-AnsibleParam $params "path" -type "path" -FailIfEmpty $true -aliases "dest", "file"
+$fragment = Get-AnsibleParam $params "fragment" -type "str" -FailIfEmpty $true -aliases "xmlstring"
+$xpath = Get-AnsibleParam $params "xpath" -type "str" -FailIfEmpty $true
+$backup = Get-AnsibleParam $params "backup" -type "bool" -Default $false
+$type = Get-AnsibleParam $params "type" -type "str" -Default "element" -ValidateSet "element", "attribute", "text"
+$attribute = Get-AnsibleParam $params "attribute" -type "str" -FailIfEmpty ($type -eq "attribute")
+
+$result = @{
+    changed = $false
+}
+
+If (-Not (Test-Path -Path $dest -PathType Leaf)){
+    Fail-Json $result "Specified path $dest does not exist or is not a file."
+}
+
+[xml]$xmlorig = Get-Content -Path $dest
+$xmlnew = $xmlorig.Clone()
+if ($type -eq "element") {
+    $xmlchild = [xml]$fragment
```

Comment:
Another try/catch here to detect if it is a valid xml fragment and return a more descriptive message if not.